io-engine/src/lvm/vg_pool.rs

```diff
@@ -0,0 +1,514 @@
+/// the capacity and free size of a given volume group
+/// sudo vgs --options=vg_size,vg_free --units=b --nosuffix --reportformat=json
+///   {
+///       "report": [
+///           {
+///               "vg": [
+///                   {"vg_name": "pool", "vg_size":"15372124160",
+/// "vg_free":"15372124160"}                ]
+///           }
+///       ]
+///   }
+#[derive(Debug, Serialize, Deserialize)]
+struct VolGroupList {
+    report: Vec<VolGroups>,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VolGroups {
+    /// Corresponds to the vg field in json output.
+    vg: Vec<VolumeGroup>,
+}
+
+/// VG query arguments, allowing filtering via --select.
+/// It's essentially a new-type wrapper over the common arguments
+/// which can't be used to build the query because... it's common
+/// between VG and LV..
+#[derive(Default, Debug)]
+pub(crate) struct QueryArgs(CmnQueryArgs);
+impl QueryArgs {
+    /// Get a comma-separated list of query selection args.
+    /// todo: should be Display trait?
+    pub(super) fn query(&self) -> String {
+        Self::query_args(&self.0)
+    }
+    /// Get a comma-separated list of query selection args.
+    pub(super) fn query_args(args: &CmnQueryArgs) -> String {
+        let mut select = String::new();
+        if let Some(vg_name) = &args.name {
+            select.push_str(&format!("vg_name={vg_name},"));
+        }
+        if let Some(vg_uuid) = &args.uuid {
+            select.push_str(&format!("vg_uuid={vg_uuid},"));
+        }
+        if let Some(vg_tag) = &args.tag {
+            select.push_str(&format!("vg_tags={vg_tag},"));
+        }
+        select
+    }
+}
+impl From<CmnQueryArgs> for QueryArgs {
+    fn from(value: CmnQueryArgs) -> Self {
+        Self(value)
+    }
+}
+
+/// An LVM Volume Group.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VolumeGroup {
+    /// Corresponds to the vg_name field in json output, the name of the
+    /// volume group.
+    #[serde(rename = "vg_name")]
+    name: String,
+    /// Corresponds to the vg_uuid field in json output, the uuid of the
+    /// volume group.
+    #[serde(rename = "vg_uuid")]
+    uuid: String,
+    /// Corresponds to the vg_size field in json output, the total capacity of
+    /// volume group in bytes.
+    #[serde(
+        deserialize_with = "super::parsing::de_number_from_string",
+        rename = "vg_size"
+    )]
+    size: u64,
+    /// Corresponds to the vg_free field in json output, the free space on
+    /// volume group in bytes.
+    #[serde(
+        deserialize_with = "super::parsing::de_number_from_string",
+        rename = "vg_free"
+    )]
+    free: u64,
+    /// Corresponds to the vg_tags field in json output, the tags set in the
+    /// volume group.
+    #[serde(
+        deserialize_with = "super::parsing::de_comma_separated",
+        rename = "vg_tags"
+    )]
+    tags: Vec<String>,
+    /// The physical vol disks used by the volume group.
+    #[serde(skip_deserializing)]
```

Comment:
Do we not need the disks? How do we compare otherwise while import that it's same disk?

Reply:
Done